File: backend/src/main/java/com/vjeditor/project/Project.java
package com.vjeditor.project;

import java.time.Instant;

public record Project(
        String id,
        String name,
        int duration,
        String aspectRatio,
        String thumbnailColor,
        Instant createdAt,
        Instant updatedAt,
        String source,
        String prompt,
        String language,
        String voice,
        String style
) {
}

File: frontend/src/models/project.ts
export type ProjectType = "blank" | "ai" | "template";

export interface Project {
  id: string;
  name: string;
  thumbnail?: string;
  duration: number;
  aspectRatio: "16:9" | "9:16" | "1:1" | "4:5";
  resolution: "720p" | "1080p" | "4K";
  type: ProjectType;
  createdAt: string;
  updatedAt: string;
}
